### PR TITLE
Jseldess/cert updates

### PR DESCRIPTION
**Pull request by jseldess**
Documented `cockroach cert` command: [create-security-certificates.md](http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/create-security-certificates.html)

And updated cert details in following docs:

- [start-a-local-cluster.md](http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/start-a-local-cluster.html)
- [use-the-built-in-sql-client.md](http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/use-the-built-in-sql-client.html) (doc for `cockroach sql`)
- [start-a-node.md](http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/start-a-node.html) (doc for `cockroach start`)
- [stop-a-node.md](http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/stop-a-node.html) (doc for `cockroach quit`)

@mberhault, let me know if I got anything wrong, missed anything, or can improve anything in any way, please.

### CODE REVIEW

File: create-security-certificates.md [hunk 1]
```diff
@@ -2,3 +2,68 @@
 title: Create Security Certificates
 toc: false
 ---
+
+A secure CockroachDB cluster uses [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) for encrypted inter-node and client-node communication and requires CA, node, and client certificates and keys. To create these certificates and keys, use the `cockroach cert` command with the appropriate subcommand and flags.
```
Comment (bdarnell) [inline create-security-certificates.md]: Some advanced users may manage their own CA, and we should say that this is OK: the `cockroach cert` command is not required. We should eventually provide guidelines on how to do this, but don't worry about it for now.

File: create-security-certificates.md [hunk 2]
```diff
@@ -2,3 +2,68 @@
 title: Create Security Certificates
 toc: false
 ---
+
+A secure CockroachDB cluster uses [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) for encrypted inter-node and client-node communication and requires CA, node, and client certificates and keys. To create these certificates and keys, use the `cockroach cert` command with the appropriate subcommand and flags. 
+
+{{site.data.alerts.callout_info}}When you create node and client certificates, you need access to a local copy of the CA certificate and key. Therefore, it's recommended to create all certificates and keys in one place and then distribute node and client certificates and keys appropriately and store the CA key in a safe place. For a walkthrough of this process, see <a href="https://www.cockroachlabs.com/docs/manual-deployment.html">Manual Deployment</a>.{{site.data.alerts.end}}
```
Comment (bdarnell) [inline create-security-certificates.md]: Generating all your keys in one place is not a best practice when it comes to security (see e.g. http://security.stackexchange.com/questions/102390/what-is-autocsr-and-is-it-secure/102408). However, we haven't implemented anything better with the `cockroach cert` tool yet (the right thing to do is to generate a key and CSR on the target node, transfer the CSR to the machine with the CA key, and get a cert that you send to the target node). I would therefore qualify the "it's recommended to" with "when using `cockroach cert`, it's recommended to".

We also need another bigger warning about the CA key: It is not recommended to copy the CA key to every node, but if you lose it you will not be able to add new nodes to your cluster. You must therefore store it somewhere safe and keep a backup.

File: create-security-certificates.md [hunk 3]
```diff
@@ -2,3 +2,68 @@
...
+#### Create the CA certificate and key
+
+~~~ shell
+$ ./cockroach cert create-ca --ca-cert=/certs/ca.cert --ca-key=/certs/ca.key 
+~~~
+
+#### Create the certificate and key for a node
+
+~~~ shell
+$ ./cockroach cert create-node nodehostname1.com nodehostname2.com --ca-cert=/certs/ca.cert --ca-key=/certs/ca.key --cert=/certs/node.cert --key=/certs/node.key
```
Comment (bdarnell) [inline create-security-certificates.md]: We often create certs with multiple hostnames for demonstration purposes, but it's a bad habit. In general, each cert should be fore a single node: `cockroach cert create-node node1.example.com --ca-cert=/certs/ca.cert --ca-key=/certs/ca.key --cert=/certs/node1.cert --cert=/certs/node1.key`

File: create-security-certificates.md (hunk 1 above)
Comment (jseldess) [inline create-security-certificates.md]: @mberhault is going to give me details about creating the equivalent certificates with openssl. Is that what you mean by users managing their own CA?

File: create-security-certificates.md (hunk 2 above)
Comment (jseldess) [inline create-security-certificates.md]: Fixing with the next commit. Thanks, @bdarnell.

File: create-security-certificates.md (hunk 3 above)
Comment (jseldess) [inline create-security-certificates.md]: Fixing with next commit.

File: create-security-certificates.md (hunk 1 above)
Comment (bdarnell) [inline create-security-certificates.md]: Yes, that's what I mean (not all users who manage their own CA will be using openssl, but they'll be comfortable translating those instructions to what they do use, unlike our opaque tool).

File: create-security-certificates.md (hunk 3 above)
Comment (mberhault) [inline create-security-certificates.md]: I believe the goal was to show that you should specify all the hostnames used to reach the given node, not specify multiple nodes.
That part is still important. For example, the beta and register clusters have certs for at least: localhost, <internal IP address>, <internal DNS>, <external IP address>, <external DNS>, beta.cockroachdb.com

Comment (jseldess) [inline create-security-certificates.md]: Ah, right, I got confused here. @bdarnell, did my initial example mislead you into thinking that we were using hostnames from two different nodes? If so, how might I reword those to be clearer?

On this point, I think it's even more important for the synopsis further up to be clarified further. Perhaps something like this?

```
# Create a node certificate and key:
$ ./cockroach cert create-node <hostname1-for-reaching-node> ... <hostnameN-for-reaching-node> --ca-cert=<path-to-ca-cert> --ca-key=<path-to-ca-key> --cert=<path-to-node-cert> --key=<path-to-node-key>
```

File: create-security-certificates.md [hunk 4]
```diff
@@ -2,3 +2,68 @@
 title: Create Security Certificates
 toc: false
 ---
+
+A secure CockroachDB cluster uses [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) for encrypted inter-node and client-node communication and requires CA, node, and client certificates and keys. To create these certificates and keys, use the `cockroach cert` command with the appropriate subcommands and flags. 
+
+When using <code>cockroach cert</code> to create node and client certificates, you will need access to a local copy of the CA certificate and key. It is therefore recommended to create all certificates and keys in one place and then distribute node and client certificates and keys appropriately. For the CA key, be sure to store it somewhere safe and keep a backup; if you loose it, you will not be able to add new nodes to your cluster. For a walkthrough of this process, see [Manual Deployment](manual-deployment.html).
```
Comment (mberhault) [inline create-security-certificates.md]: s/"if you loose it"/"if you lose it"/

s/"you will not be able to add new nodes"/"you will not be able to add new nodes or clients"/

File: create-security-certificates.md [hunk 5]
```diff
@@ -2,3 +2,68 @@
...
+
+<div id="toc"></div>
+
+## Subcommands
+
+Subcommand | Usage 
+-----------|------
+`create-ca` | Create the Certificate Authority (CA) certificate and key for the entire cluster.
```
Comment (mberhault) [inline create-security-certificates.md]: we should mention that it's a "self-signed" CA.

File: create-security-certificates.md (hunk 4 above)
Comment (jseldess) [inline create-security-certificates.md]: Good catches. Fixed with next commit.

File: create-security-certificates.md (hunk 5 above)
Comment (jseldess) [inline create-security-certificates.md]: Fixing with next commit.

**Comment by jseldess** [comment]
@mberhault, @bdarnell, updated the docs to be more clear that the cert for a node can list multiple address for that node, not for other nodes. 

Marc, when you send over realistic hostname examples and openssl instructions, I'll update further.

**Comment by jseldess** [comment]
Merging. Will open an issue to update `create-node` example with realistic hostnames.

**Pull request merged**